src/components/pages/Authentication/hooks.ts

```diff
@@ -1,24 +1,28 @@
-import { useNavigate } from "@reach/router";
+import { useLocation, useNavigate } from "@reach/router";
 import axios from "axios";
 import { useCallback, useEffect } from "react";
-import { useIntl } from "react-intl";
 
 import { useAuth, useCleanUrl } from "@reearth/auth";
 import { useTeamsQuery } from "@reearth/gql";
 import { useTeam, useNotification } from "@reearth/state";
 
 export default () => {
-  const intl = useIntl();
-  const { isAuthenticated, isLoading, error: authError, logout } = useAuth();
+  const { isAuthenticated, isLoading, error: authError, logout, login } = useAuth();
   const error = useCleanUrl();
   const navigate = useNavigate();
+  const location = useLocation();
   const [currentTeam, setTeam] = useTeam();
   const [, setNotification] = useNotification();
   const passwordPolicy = window.REEARTH_CONFIG?.passwordPolicy;
 
   const { data, loading } = useTeamsQuery({ skip: !isAuthenticated });
   const teamId = currentTeam?.id || data?.me?.myTeam.id;
 
+  useEffect(() => {
+    if (location.pathname === "/login" && !new URLSearchParams(window.location.search).has("id"))
+      login();
+  }, [login, location.pathname]);
+
```

Comment:
I believe I'm doing something similar in [this code block](https://github.com/reearth/reearth-web/blob/d824b61295603fd4348c29477ab0c8b88f799ed0/src/components/organisms/Authentication/RootPage/hooks.ts#L49-L62). Unless I am wrong and there is a reason for this, it might cause issues having both.

Reply:
The reason for this is when user navigate to `/login` without an id, it will re-generate the id, as requested recently on Notion.